### PR TITLE
Updates for Julia 0.6, drop 0.4 support

### PR DESCRIPTION
**Pull request by ararslan**
This PR removes unnecessary `@compat`s, adds some necessary `@compat`s, drops support for 0.4, and facilitates support for 0.6.

### CODE REVIEW

**Comment by ararslan** [comment]
Current 0.6 failure due to syntax that doesn't parse in IniFile. That should be fixed once https://github.com/JuliaLang/METADATA.jl/pull/8299 is merged. I think most (all?) of the current depwarns on 0.6 are now due to Cairo, IniFile, and Winston.

File: REQUIRE [hunk 1]
```diff
@@ -1,5 +1,5 @@
-julia 0.4
-Compat 0.7.20
+julia 0.5
+Compat 0.18.0
```
Comment (tkelman) [inline REQUIRE]: 0.17

Comment (ararslan) [inline REQUIRE]: Right, thanks

**Comment by ararslan** [comment]
Winston won't work without this change—the `get_args` update in particular—and the tests here won't pass without Winston working... Not sure what the best course of action is.

**Comment by ararslan** [comment]
Woo! Tests passing now! I had to comment out the Winston tests temporarily; I figured once Winston supports 0.6 we can uncomment. Does that seem reasonable?

**Comment by timholy** [comment]
Yes. Thanks for tackling this!

**Comment by ararslan** [comment]
My pleasure! Is this good to go now? I think I've dealt with everything I had set out to deal with.

Unrelated question: Do you know what's keeping this package from being able to precompile?

**Comment by timholy** [comment]
> Do you know what's keeping this package from being able to precompile?

Nope.

**Comment by ararslan** [comment]
Haha alright. Any chance you'd be willing to tag a new release?

**Pull request merged**